pkg/resourcemonitor/resourcemonitor.go

```diff
@@ -259,6 +257,28 @@ func (rm *resourceMonitor) updateNodeCapacity() error {
 	return nil
 }
 
+func (rm *resourceMonitor) resUpdated(old, new interface{}) {
+	nOld := old.(*v1.Node)
+	nNew := new.(*v1.Node)
+
+	if nNew.Name != rm.nodeName {
+		return
+	}
+
+	// the status frequency update are configurable via the node-status-update-frequency option in Kubelet
+	if !reflect.DeepEqual(nOld.Status.Capacity, nNew.Status.Capacity) ||
+		!reflect.DeepEqual(nOld.Status.Allocatable, nNew.Status.Allocatable) {
+		klog.V(4).Infof("update node resources")
+		if err := rm.updateNodeCapacity(); err != nil {
+			klog.ErrorS(err, "error while updating node capacity")
+		}
+		if err := rm.updateNodeAllocatable(); err != nil {
+			klog.ErrorS(err, "error while updating node allocatable")
+		}
+		rm.updateDevicesCapacity()
```

Comment:
We can simply call `updateNodeResources` here instead of reimplementing that logic here.

Reply:
good catch, this seems like a good idea

Reply:
The reason why I didn't do that is that we can't return the error from ```updateNodeResources```  and it would be more clear if we would be able to log explicitly from which of the functions the error came from.

Reply:
what about:
```
func (rm *resourceMonitor) updateNodeResources() error {
	if err := rm.updateNodeCapacity(); err != nil {
		return fmt.Errorf("error while updating node capacity: %w", err)
	}
	if err := rm.updateNodeAllocatable(); err != nil {
		return fmt.Errorf("error while updating node allocatable: %w", err)
	}
	rm.updateDevicesCapacity()
	return nil
}

...
	klog.V(3).Infof("update node resources")
	if err := rm.updateNodeResources(); err != nil {
		klog.ErrorS(err, "while updating node resources")
	}
...

```

Reply:
Nice!

Reply:
```
what about:

func (rm *resourceMonitor) updateNodeResources() error {
	if err := rm.updateNodeCapacity(); err != nil {
		return fmt.Errorf("error while updating node capacity: %w", err)
	}
	if err := rm.updateNodeAllocatable(); err != nil {
		return fmt.Errorf("error while updating node allocatable: %w", err)
	}
	rm.updateDevicesCapacity()
	return nil
}

...
	klog.V(3).Infof("update node resources")
	if err := rm.updateNodeResources(); err != nil {
		klog.ErrorS(err, "while updating node resources")
	}
...
```

Exactly what I was trying to suggest. Thanks @fromanirh.